src/modules/blob.js

```diff
@@ -0,0 +1,86 @@
+import { encodeAddress, randomAsHex } from '@polkadot/util-crypto';
+import { getSignatureFromKeyringPair, getStateChange } from '../utils/misc';
+import NoBlobError from '../utils/errors/no-blob-error';
+
+export const DockBlobMethod = 'dock';
+export const DockBlobQualifier = `blob:${DockBlobMethod}:`;
+export const DockBlobIdByteSize = 32;
+
+/**
+ * Create and return a fully qualified Dock Blob, i.e. "did:dock:<SS58 string>"
+ * @returns {string} - The Blob
+ */
+export function createNewDockBlobId() {
+  const hexId = randomAsHex(DockBlobIdByteSize);
+  return blobHexIdToQualified(hexId);
+}
+
+/**
+ * Return a fully qualified Dock Blob, i.e. "did:dock:<SS58 string>"
+ * @returns {string} - The Blob
+ */
+export function blobHexIdToQualified(hexId) {
+  const ss58Id = encodeAddress(hexId);
+  return `${DockBlobQualifier}${ss58Id}`;
+}
+
+/** Class to create and update Blobs on chain. */
+class BlobModule {
+  /**
+   * Creates a new instance of BlobModule and sets the api
+   * @constructor
+   * @param {object} api - PolkadotJS API Reference
+   */
+  constructor(api) {
+    this.api = api;
+    this.module = api.tx.blobStore;
+  }
+
+  /**
+   * Register a new Blob on the Dock Chain
+   * @param {object} blob - struct to store on chain
+   * @param {object} keyPair - Key pair to sign with
+   * @return {object} The extrinsic to sign and send.
+   */
+  new(blob, keyPair) {
+    const serializedBlob = this.getSerializedBlob(blob);
+    const signature = getSignatureFromKeyringPair(keyPair, serializedBlob);
+    return this.module.new(blob, signature.toJSON());
+  }
+
+  /**
+   *  Get blob with given id from the chain. Throws if the blob can't be found.
+   * @param {string} id - Can either be a full blob id like blob:dock:0x... or just the hex identifier
+   * @returns {Promise<Array>} - A 2-element array where the first is the author and the second is the blob contents.
+   */
+  async getBlob(id) {
+    id.replace(DockBlobQualifier, '');
```

Comment:
`replace` doesn't mutate given string but returns a new string. Secondly why not just `id = id.slice(DockBlobQualifier)`, its faster and does exactly what needs to be done. You don't need to search in the whole string. What if the qualifier was added to the hex id and not ss58? Have a look at the `getDocument` for did module.